bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Image.java

```diff
@@ -2256,15 +2257,19 @@ public void setBackground(Color color) {
 	device.internal_dispose_GC(hDC, null);
 }
 
-private void setCurrentDeviceZoom(int newZoomFactor) {
-	if (this.currentDeviceZoom != newZoomFactor) {
-		this.currentDeviceZoom = newZoomFactor;
+private void setCurrentNativeZoom(int newZoomFactor) {
+	if (this.getZoom() != newZoomFactor) {
```

Comment:
Can you explain how does this work? Specifically: why do you now compare using `getZoom()` instead of the current device zoom (it was `currentDeviceZoom` before).
It seems a bit odd that ones compares against the output of a method instead of the previously stored value but I assume it has something to do with rounding errors?

Also, please rename the parameter to `deviceZoom` since it's going to end up being stored in `this.currentDeviceZoom` (make the names "match (better)")

Reply:
It was current device zoom earlier and now it is current NATIVE zoom. getZoom method converts native zoom to device zoom. For image's internal usage only device zoom is relevant. In this case we only wanna create a new handle for the image when the device zoom chnages (i.e. if the native zoom changes from 100 to 125, the device zoom would still stay 100, in this case we would end up creating 2 handles for 100 whioch is inefficient). Also the setCurrentNativeZoom sets the height and width of image to -1, making it need to call init method again to be usable. 

For this PR, this is sufficient since in the later GC and Image PRs, we would extend to the current native zoom to provide the native zoom to GCData when creating a new GC for a drawable of type Image. In further Image PR, you'd see this behaviour changing because then we would not need to set current native zoom anymore for the image but rather initial native zoom only.

Reply:
Still the name `zoomFactor` seems to be inappropriate. And assigning `getZoom()` to some `oldZoom` variable would make the code more comprehensible.